### DISCUSSION TITLE
Datamodule without Trainer (for inference)

### DISCUSSION THREAD

**Discussion by indigoviolet** [Lightning-AI/pytorch-lightning#6502] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
In my usage, LightningDatamodule is currently encapsulating batch collation, moving to device, and batch transformations (via `on_after_batch_transfer`). 

However, when I want to do inference on a bunch of inputs, I want the same steps to happen. What is the recommended way to achieve this? The problem is that Trainer drives the device transfers and hooks around it, and I don't have a Trainer during inference.

**Comment by carmocca** [comment, marked as answer]
Why would you not want to use the Trainer?

You can now use `trainer.predict` for inference (will be in beta after the 1.3 release)
